src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,154 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T item) {
+            this.item = item;
+        }
+
+    }
+
+    public void insertAtTheBeginningOfTheList(T value) {
+        Node<T> newNode = new Node<>(value);
+        head.prev = newNode;
+        newNode.next = head;
+        head = newNode;
+    }
+
+    public void insertAtTheEndOfTheList(T value) {
+        Node<T> newNode = new Node<>(value);
+        tail.next = newNode;
+        newNode.prev = tail;
+        tail = newNode;
+    }
+
+    public void insertAtTheMiddleOfTheList(T value, int index) {
+        Node<T> newNode = new Node<>(value);
+        Node<T> currentNode = head;
+        for (int i = 0; i < index; i++) {
+            currentNode = currentNode.next;
+        }
```

Comment:
[checklist:](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html) Don’t iterate your nodes from the head to tail or vice versa. Lets move logic of finding node by index in separate method as we use it several times in the code